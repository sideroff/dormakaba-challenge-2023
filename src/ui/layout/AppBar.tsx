import { useState, useCallback, MouseEvent } from 'react';
import MuiAppBar from '@mui/material/AppBar';
import Container from '@mui/material/Container';
import Toolbar from '@mui/material/Toolbar';
import { AppBarLogo } from '@/ui/layout/AppBarLogo';
import { XSScreenMenu } from '@/ui/layout/XSScreenMenu';
import { MDScreenMenu } from '@/ui/layout/MDScreenMenu';
import { AppBarDate } from '@/ui/layout/AppBarDate';

export function AppBar() {
  const [navAnchorElement, setNavAnchorElement] = useState<null | HTMLElement>(
    null,
  );

  const handleNavMenuOpen = useCallback((event: MouseEvent<HTMLElement>) => {
    setNavAnchorElement(event.currentTarget);
  }, []);

  const handleNavMenuClose = useCallback(() => {
    setNavAnchorElement(null);
  }, []);

  return (
    <MuiAppBar position="static">
      <Container>
        <Toolbar disableGutters>
          <AppBarLogo />

          <XSScreenMenu
            navAnchorElement={navAnchorElement}
            onNavMenuOpen={handleNavMenuOpen}
            onNavMenuClose={handleNavMenuClose}
          />

          <MDScreenMenu onNavMenuClose={handleNavMenuClose} />

          <AppBarDate />
        </Toolbar>
      </Container>
    </MuiAppBar>
  );
}
