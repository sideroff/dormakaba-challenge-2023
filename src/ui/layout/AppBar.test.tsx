import { render } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import { AppBar } from './AppBar';

jest.mock('next/router', () => require('next-router-mock'));

describe('AppBar', () => {
  beforeAll(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date('2023-02-12T14:30:20.124'));
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should render correctly', () => {
    mockRouter.push('/test');

    const { container } = render(<AppBar />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
