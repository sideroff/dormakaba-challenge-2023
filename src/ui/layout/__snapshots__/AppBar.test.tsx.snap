// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should render correctly 1`] = `
<header
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
    >
      <div
        class="MuiBox-root css-8eu3pw"
      >
        <a
          href="/"
        >
          <img
            alt="dormakaba logo"
            data-nimg="1"
            decoding="async"
            height="30"
            loading="lazy"
            src="/_next/image?url=%2Flogo.png&w=640&q=75"
            srcset="/_next/image?url=%2Flogo.png&w=384&q=75 1x, /_next/image?url=%2Flogo.png&w=640&q=75 2x"
            style="color: transparent;"
            width="269"
          />
        </a>
      </div>
      <div
        class="MuiBox-root css-1lvtzne"
      >
        <button
          aria-controls="menu-appbar"
          aria-haspopup="true"
          aria-label="menu button"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <ul
        class="MuiList-root MuiList-padding css-hmqy6w-MuiList-root"
      >
        <a
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-72rlw3-MuiButtonBase-root-MuiListItemButton-root"
          href="/doors"
          tabindex="0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Doors
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </ul>
      <p
        class="MuiTypography-root MuiTypography-body1 css-p1x2na-MuiTypography-root"
      >
        2/12/2023
      </p>
    </div>
  </div>
</header>
`;
